src/main/java/com/amazon/opendistroforelasticsearch/ad/rest/handler/ValidateAnomalyDetectorActionHandler.java

```diff
@@ -0,0 +1,590 @@
+        }
+        return featureFields;
+    }
+
+    private synchronized void samplingHelper(long[][] timeRanges, long detectorInterval) throws IOException {
+        inferAgain.set(false);
+        List<String> featureFields = getFeatureFieldNames();
+        MultiSearchRequest sr = new MultiSearchRequest();
+        for (int i = 0; i < NUM_OF_INTERVAL_SAMPLES; i++) {
+            long rangeStart = timeRanges[i][0];
+            long rangeEnd = timeRanges[i][1];
+            RangeQueryBuilder rangeQueryRandom = new RangeQueryBuilder(anomalyDetector.getTimeField())
+                .from(rangeStart)
+                .to(rangeEnd)
+                .format("epoch_millis")
+                .includeLower(true)
+                .includeUpper(false);
+            BoolQueryBuilder qb = QueryBuilders.boolQuery().filter(rangeQueryRandom).filter(anomalyDetector.getFilterQuery());
+            for (int j = 0; j < featureFields.size(); j++) {
+                qb.filter(QueryBuilders.existsQuery(featureFields.get(j)));
+            }
+            SearchSourceBuilder internalSearchSourceBuilder = new SearchSourceBuilder().query(qb).size(1).terminateAfter(1);
+            sr.add(new SearchRequest(anomalyDetector.getIndices().get(0)).source(internalSearchSourceBuilder));
+        }
+        client.multiSearch(sr, ActionListener.wrap(searchResponse -> {
+            if (doneInferring(detectorInterval, searchResponse)) {
+                checkWindowDelay();
+            }
+        }, exception -> onFailure(exception)));
+    }
+
+    private synchronized boolean doneInferring(long detectorInterval, MultiSearchResponse searchResponse) {
+        long originalInterval = Optional
+            .ofNullable((IntervalTimeConfiguration) anomalyDetector.getDetectionInterval())
+            .map(t -> t.toDuration().toMillis())
+            .orElse(0L);
+        final AtomicInteger hitCounter = new AtomicInteger();
+        for (MultiSearchResponse.Item item : searchResponse) {
+            SearchResponse response = item.getResponse();
+            if (response.getHits().getTotalHits().value > 0) {
+                hitCounter.incrementAndGet();
+            }
+        }
+        inferAgain.set(true);
+        notify();
+        if (hitCounter.doubleValue() / (double) NUM_OF_INTERVAL_SAMPLES < SAMPLE_SUCCESS_RATE) {
+            if ((detectorInterval * INTERVAL_RECOMMENDATION_MULTIPLIER) >= MAX_INTERVAL_LENGTH) {
+                suggestedChangesMap
+                    .computeIfAbsent("detection_interval", k -> new ArrayList<>())
+                    .add("detector interval: failed to infer max up too: " + MAX_INTERVAL_LENGTH);
+            } else {
+                return false;
+            }
+        } else if (detectorInterval != originalInterval) {
+            suggestedChangesMap.computeIfAbsent("detection_interval", k -> new ArrayList<>()).add(Long.toString(detectorInterval));
+            inferAgain.set(false);
+            return true;
```

Comment:
You may not need those 2 lines as they are same as below

Reply:
This is needed for the case when the interval is above the MAX_INTERVAL_LENGTH, in this case the infer again should be changed too false from true and the return should be true.

Reply:
If code execution goes into block of `if{}` at line 622, the next step from that block is line 629/630, which is same last 2 steps of `if{}` from line 622, aka line 626/627. That is why line 626/627 can be removed from my understanding.

Reply:
done, I missed this before when reviewing for the first time.